peers/peerpool.go

```diff
@@ -107,8 +107,8 @@ func (p *PeerPool) setDiscoveryTimeout() {
 }
 
 // Start creates topic pool for each topic in config and subscribes to server events.
-func (p *PeerPool) Start(server *p2p.Server) error {
-	if server.DiscV5 == nil {
+func (p *PeerPool) Start(server *p2p.Server, discovery Discovery) error {
```

Comment:
Does it make sense to make `Discovery` a part of `PeerPool` structure? Seems like it will simplify the rest of the code (less arguments passing to functions), but maybe I'm missing something here.

Reply:
moved